design/ceph/block-ceph-delete-if-pool-data.md

```diff
@@ -0,0 +1,146 @@
+    - `radosgw-admin --rgw-zonegroup=$zonegroup --rgw-zone=$zone bucket list` to get buckets
+    - Ignore a single instance of a bucket named `rook-ceph-bucket-checker-*`
+    - If other buckets exist, assume the store is in-use
+
+##### Corner cases
+1. If a user creates an object store without using CephObjectStore and inside that store creates a
+bucket with a name in the format `rook-ceph-bucket-checker-*`, Rook will not stop cluster deletion.
+2. `.rgw.root` is present in Ceph pool list but not included in info from `radosgw-admin`... is this
+   item ALWAYS the same name?????? (TODO)
+
+##### Background technical information
+Zone name and zonegroup name are usually the same as CephObjectStore name except with new
+multi-site setups. Because of this multi-site case, we have to assume they are different.
+
+Zones can be effectively thought of as the "object store" itself. The name of the CephObjectStore
+is actually irrelevant.
+
+Zones exist in a zone group, and the zone group is also necessary to list buckets.
+
+Zone groups also exist in a realm, but the realm doesn't seem to be necessary to list buckets.
+
+A `rook-ceph-bucket-checker-...` bucket is always present in the object stores created by Rook. Rook
+periodically writes, reads, and deletes test information in the object store check that the store is
+healthy.
+
+### CephNFS / nfs-ganesha
+TODO: Do we need to do anything special here since this uses user-created or Rook-created pools?
+
+### CephBlockStores / RBD
+For any pools that are not part of a filesystem or object store, we should assume these pools are
+block storage pools. It is simple to determine whether a pool is storing data with `ceph df`.
+
+The only special consideration is that Rook should ignore the `device_health_metrics` pool.
+
+TODO: how can we keep on top of pools that we should ignore like `device_health_metrics` and
+`.rgw.root`? One answer: GOOD INTEGRATION TESTING OF THIS SCENARIO!
+
+
+## Additional thoughts
+
+### Checking for the existence of Persistent Volumes
+With this design in place, it is no longer necessary to check for the existence of Persistent
+Volumes (PVs) which are backed by a CephCluster when it is being deleted. Existence of pools is a
+prerequisite for a PV to exist. However, if a PV exists which references a pool, this design
+proposes to consider this an active "intent to use" even if the application claiming the PV has not
+written any data.
+
+### Pool information in CR statuses
+For Rook resources that create pools, it would be good to list the pools on the CR status to help
+admins cross-referencing things during deletion (but also during normal Day 1 and Day 2 ops). List
+data and metadata pools separately. E.g., look at a CephFilesystem CR's status.pools.data to find a
+data pool for CephNFS to use (day 1).
+
+### CephClusters in external mode
+TODO: should we act differently in this case since the cluster exists merely to connect to an
```

Comment:
The deletion will:

* affect CSI, since the cm might go away (TBV)
* affect OB/OBC since the CephObjectStore relies on the existence of CephCluster in some place IIRC